service/history/hsm/tree.go

```diff
@@ -121,12 +123,17 @@ type Node struct {
 	persistence *persistencespb.StateMachineNode
 	definition  StateMachineDefinition
 	backend     NodeBackend
+	tombstones  TombstoneLog // Only used at root
```

Comment:
Just to clarify, are you saying instead of having a separate TombstoneLog, we should have a single operation log that tracks all operations. Note that we still only need to track deletions (since they supersede transitions), and still use the optimization where parent deletions make child operations unnecessary. Is that right?

Reply:
I would just use a single mechanism to track both transitions and deletions and change the way the `Outputs` function works.

Reply:
Done